### ISSUE TITLE
New "Builder" role.

### ISSUE THREAD

**Issue by carletex** [moonshotcollective/scaffold-directory#97]
Right now, we have the following roles:

```js
export const USER_ROLES = {
  admin: "user_role.administrator",
  anonymous: "user_role.anonymous",
  registered: "user_role.registered",
};
```

**admin**: "isAdmin": true on the database.
**anonymous**: the user hasn't connected their wallet || the user address is NOT on the database
**registered**: the user address is on the database.

We need a new **builder** role =  registered user that has completed 1 or 2 challenges. These builders will be able to see the build list + submit new ones (with admin approval)

**Database Changes**
We could add `"isBuilder": true` as a quick fix, but a more scalable solution would be to add a `role` property. I'd go with the latter since we might add more roles in the future. We will need to migrate admins: remove `isAdmin` prop and create `role: 'user_role.administrator'`

From Austin:
> I think letting people submit/propose repo urls to add to the “builds” list. We can coach them in the submission form to make a good readme and then they can turn the repo in as a build. I think that probably leads to a new “builder” role on the site that you get after completing a challenge or two. This builder role unlocks all the cool views like builds, activity feed, etc. (I think activity feed is fun next priority too. Getting to see people submit builds and complete challenges sounds like a lot of fun)

Also:
> Roles could possibly be these nontransferrable NFTs we are building

Since we don't have the NFTs integration yet (see #95), I'd go with the database implementation and we can switch to reading roles from polygon at some point.

**Comment by dgrcode** [comment]
How do you think roles should evolve?

For example, when we give someone the builder role, should we remove the registered? or we just add the builder role? Would it be possible that someone has a builder role but not the registered role?

I'm just thinking that having an array could be an overhead that doesn't add much value. We could just store the highest role someone has, and then have a utility to check if the user is "at least" builder. The hierarchy of roles would depend on the logic we implement inside the utility.

What do you think?

**Comment by carletex** [comment]
@dgrcode I agree.

I think in this case higher roles will just be super sets of lower roles.

Let's go with a single value. Let me update the initial comment

**Comment by carletex** [comment]
Fixed on #101
